functions/Get-DbaDatabase.ps1

```diff
@@ -313,15 +313,19 @@ function Get-DbaDatabase {
                 $lastCopyOnlyBackups = Get-DbaDbBackupHistory -SqlInstance $server -LastFull -IncludeCopyOnly | Where-Object IsCopyOnly
                 if ($NoFullBackupSince) {
                     $lastFullBackups = $lastFullBackups | Where-Object End -gt $NoFullBackupSince
+                    $lastCopyOnlyBackups = $lastCopyOnlyBackups | Where-Object End -gt $NoFullBackupSince
                 }
-                $inputObject = $inputObject | Where-Object { $_.Name -cnotin $lastFullBackups.Database -and $_.Name -ne 'tempdb' }
+
+                $hasCopyOnly = $inputObject | Where-CollationSensitiveObject -Property Name -In -Value $lastCopyOnlyBackups.Database -Collation $server.Collation
+                $inputObject = $inputObject | Where-CollationSensitiveObject -Property Name -Notin -Value $lastFullBackups.Database -Collation $server.Collation
+                $inputObject = $inputObject | Where-Object Name -cne 'tempdb'
             }
             if ($NoLogBackup -or $NoLogBackupSince) {
-                $lastLogBackups = Get-DbaDbBackupHistory -SqlInstance $server -LastLog
-                if ($NoLogBackupSince) {
-                    $lastLogBackups = $lastLogBackups | Where-Object End -gt $NoLogBackupSince
+                if (!$NoLogBackupSince) {
+                    $NoLogBackupSince = New-Object -TypeName DateTime
+                    $NoLogBackupSince = $NoLogBackupSince.AddMilliSeconds(1)
                 }
-                $inputObject = $inputObject | Where-Object { $_.Name -cnotin $lastLogBackups.Database -and $_.Name -ne 'tempdb' -and $_.RecoveryModel -ne 'Simple' }
+                $inputObject = $inputObject | Where-Object { $_.LastLogBackupDate -lt $NoLogBackupSince -and $_.Name -ne 'tempdb' -and $_.RecoveryModel -ne 'Simple' }
```

Comment:
I think we never needed `-and $_.Name -ne 'tempdb'` as tempdb is always in recovery model simple.